sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/OperationPoliciesTest.java

```diff
@@ -0,0 +1,772 @@
+            createBulkOptions(operationType, spanName, cosmosRequestOptions);
+            createChangeFeedOptions(spanName, cosmosRequestOptions);
+            cosmosOperationDetails.setCommonOptions(cosmosRequestOptions);
+        };
+
+        CosmosClientBuilder[] clientBuilders = new CosmosClientBuilder[3];
+        clientBuilders[0] =  new CosmosClientBuilder().endpoint(TestConfigurations.HOST)
+            .credential(credential)
+            .gatewayMode()
+            .addOperationPolicy(policy);
+        clientBuilders[1] =  new CosmosClientBuilder().endpoint(TestConfigurations.HOST)
+            .credential(credential)
+            .addOperationPolicy(policy);
+        clientBuilders[2] = new CosmosClientBuilder().endpoint(TestConfigurations.HOST)
+            .credential(credential)
+            .addOperationPolicy((cosmosOperationDetails) -> {
+                CosmosDiagnosticsContext cosmosDiagnosticsContext = cosmosOperationDetails.getDiagnosticsContext();
+                String operationType = cosmosDiagnosticsContext.getOperationType();
+                String spanName = cosmosDiagnosticsContext.getSpanName();
+                CosmosRequestOptions cosmosRequestOptions = new CosmosRequestOptions();
+                createReadDeleteBatchEtcOptions(operationType, spanName, cosmosRequestOptions);
+                createQueryReadAllItemsOptions(operationType, spanName, cosmosRequestOptions);
+                createReadManyOptions(spanName, cosmosRequestOptions);
+                cosmosOperationDetails.setCommonOptions(cosmosRequestOptions);
+            }).addOperationPolicy((cosmosOperationDetails) -> {
+                CosmosDiagnosticsContext cosmosDiagnosticsContext = cosmosOperationDetails.getDiagnosticsContext();
+                String operationType = cosmosDiagnosticsContext.getOperationType();
+                String spanName = cosmosDiagnosticsContext.getSpanName();
+                CosmosRequestOptions cosmosRequestOptions = new CosmosRequestOptions();
+                createBulkOptions(operationType, spanName, cosmosRequestOptions);
+                createChangeFeedOptions(spanName, cosmosRequestOptions);
+                cosmosOperationDetails.setCommonOptions(cosmosRequestOptions);
+            });
+        return clientBuilders;
+    }
+
+    @BeforeClass(groups = {"fast"}, timeOut = SETUP_TIMEOUT)
+    public void before_OperationPoliciesTest() {
+        assertThat(this.client).isNull();
+        this.client = getClientBuilder().buildAsyncClient();
+        CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.client);
+        container = client.getDatabase(asyncContainer.getDatabase().getId()).getContainer(asyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"fast"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        assertThat(this.client).isNotNull();
+        this.client.close();
```

Comment:
You're inheriting `TestSuiteBase` which provides `safeClose` which does the same (does null check prior as well).

Reply:
changed to use safeClose()